doc/languages-frameworks/javascript.section.md

```diff
@@ -439,9 +438,15 @@ This script by default runs `yarn --offline build`, and it relies upon the proje
 - `yarnBuildScript`: Sets a different `yarn --offline` subcommand (defaults to `build`).
 - `yarnBuildFlags`: Single string list of additional flags to pass the above command, or a Nix list of such additional flags.
 
+#### `yarnInstallHook` arguments {#javascript-yarninstallhook}
+
+To install the package `yarnInstallHook` uses both `npm` and `yarn` to cleanup project files and dependencies. To disable this phase, you can set `dontYarnInstall = true` or override the `installPhase`. Below is a list of additional `mkDerivation` arguments read by this hook:
+
+- `dontYarnPrune`: Disables the removal of devDependencies from `node_modules` before installation.
```

Comment:
I am still confused regarding this... https://yarnpkg.com/cli/install doesn't mention the word `devDependencies`, or the word prune. It'd also be nice to find a single example where this argument is needed, and explain like that why is that.

Reply:
Here ya go: 
https://classic.yarnpkg.com/en/docs/cli/prune
https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-production-true-false

> It'd also be nice to find a single example where this argument is needed, and explain like that why is that.

I'm not sure what you mean here. Is it:

A) "dontYarnPrune doesn't seem very useful, we should always prune"? 

I left the option to disable pruning because I don't trust that yarn will not suffer from the same problem as already exists in npm — not being able to prune without internet access sometimes.

Like other cleanup steps (stripping binaries, for example), this feature is nice to have, but imprudent to not allow others to easily disable it.

B) "prune doen't seem very useful, we should not do it"

```sh
$ du -sh result-not-pruned/ result-pruned/
702M	result-not-pruned/
120M	result-pruned/
```

Reply:
What confuses me is only the terminology you use for the different `dontYarn*` variables, and not the logic you implement. 

> > It'd also be nice to find a single example where this argument is needed, and explain like that why is that.
> 
> I'm not sure what you mean here. Is it:

Indeed my phrasing wasn't clear. What I meant was: Give me 1 example for a package that needs to set `dontYarnPrune = true`, and show me how the `yarn install` command fails - with which errors, and maybe I'll get convinced that it'd be more correct to disable that `yarn install` command with a variable named `dontYarnPrune`, and not `dontYarnInstall`. Both of your A & B arguments you gave are very convincing, but still don't convince me that the naming is good.

> Here ya go:
> [classic.yarnpkg.com/en/docs/cli/prune](https://classic.yarnpkg.com/en/docs/cli/prune)
> [classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-production-true-false](https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-production-true-false)

Again, your shell code links the shell variable `dontYarnPrune` to the command `yarn install`. Hence I seek to find a connection / relation between the functionality of the `yarn install` command, and the word "prune". The 1st link is not the documentation of the command `yarn install` which you use right after the `dontYarnPrune` check, and the 2nd link doesn't mention the word prune.

Reply:
> Again, your shell code links the shell variable `dontYarnPrune` to the command `yarn install`. Hence I seek to find a connection / relation between the functionality of the `yarn install` command, and the word "prune". The 1st link is not the documentation of the command `yarn install` which you use right after the `dontYarnPrune` check, and the 2nd link doesn't mention the word prune.

Npm has a command called `prune`: https://docs.npmjs.com/cli/v10/commands/npm-prune

In Yarn, the functionality of this command is bundled in the `install` command. You can see this in the first link "The prune command isn’t necessary. yarn install will prune extraneous packages.", meaning the functionality of a `prune` command is bundled in the `install` command.

I think my usage of `yarn install --production=true` is the closes possible to what `yarn prune` would do, if it was a separate command (right now it just returns an error message saying you should use `yarn install`).

Reply:
> I think we're talking about the same thing on two threads 😃

Correct :).

> Npm has a command called `prune`: [docs.npmjs.com/cli/v10/commands/npm-prune](https://docs.npmjs.com/cli/v10/commands/npm-prune)
> 
> In Yarn, the functionality of this command is bundled in the `install` command. You can see this in the first link "The prune command isn’t necessary. yarn install will prune extraneous packages.", meaning the functionality of a `prune` command is bundled in the `install` command.

OK I see that now. I missed that text when I opened the link last time.

> I think my usage of `yarn install --production=true` is the closes possible to what `yarn prune` would do,

Hmm OK I see. After testing building `postlight-master` with and without `dontYarnPrune` set and I see a difference that can be explained by the `--production=true` flag, and I _am_ convinced that we should run `yarn install --production=true` to reduce the amount of files in the `node_modules` directory.

What if we'd rename the `dontYarnPrune` variable to `yarnIncludeDevDeps`? This way you'd be more aligned with upstream's terminology.

Reply:
> What if we'd rename the `dontYarnPrune` variable to `yarnIncludeDevDeps`? This way you'd be more aligned with upstream's terminology.

That sound alright, maybe even even `yarnKeepDevDeps`, since they will always be installed, and then optionally removed?

Reply:
> > What if we'd rename the `dontYarnPrune` variable to `yarnIncludeDevDeps`? This way you'd be more aligned with upstream's terminology.
> 
> That sound alright, maybe even even `yarnKeepDevDeps`, since they will always be installed, and then optionally removed?

Sounds even better.